Paws4Me/Paws4Me/Adoption Form/Adoption Form Page 1/View Controller/AdoptFirstFormPageViewController.swift

```diff
@@ -0,0 +1,218 @@
+//
+//  AdoptFirstFormPageViewController.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/04/01.
+//
+
+import Foundation
+import UIKit
+
+class AdoptFirstFormPageViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+
+    // : MARK: - IBOutlet
+    @IBOutlet weak private var adoptFirstFormTableView: UITableView?
+    @IBOutlet weak private var progress1: UIView!
+    @IBOutlet weak private var progress3: UIView!
+    @IBOutlet weak private var progress2: UIView!
+    @IBOutlet weak private var progress4: UIView!
+
+    // : MARK: - Var/Lets
+    private lazy var firstFormViewModel = FirstFormViewModel()
+    private var textFields = [UITextField]()
+    private var uiSwitches = [UISwitch]()
+    private var indexTextfield = 0
+    private var indexUISwitch = 0
+
+    // : MARK: - Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpTableView()
+        setUpProgress()
+    }
+
+    // : MARK: - IBAction
+    @IBAction func form2NextButton(_ sender: Any) {
+
+        var indexTextfield = 0
+
+        while textFields.count != indexTextfield {
+            if let textField = self.textFieldForTag( tag: indexTextfield ) {
+                guard let textfieldValue = textField.text else { return }
+                var cellData = firstFormViewModel.arrayForm.filter({$0.cellIndex == indexTextfield})
+                cellData[0].data?.itemValue = textfieldValue
+                print( cellData[0] )
+                indexTextfield += 1
+            }
+    }
+        var indexSwitch = 0
+        while uiSwitches.count != indexSwitch {
+            if let uiSwitch = self.uiSwitchForTag(tag: indexSwitch) {
+                var cellData = firstFormViewModel.arrayForm.filter({$0.cellIndex == indexSwitch})
+                cellData[0].data?.itemValue = uiSwitch.isOn.description
+                print( cellData[0] )
+                indexSwitch += 1
+            }
+    }
+    }
+
+    // : MARK: - Functions
+   private func setUpProgress() {
+        progress1.makeCircle()
```

Comment:
Please research outlet groups and see if we can improve this code 😄

Reply:
Will do so 👍🏻